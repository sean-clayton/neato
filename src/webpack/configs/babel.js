// @flow

import path from 'path'
import { HotModuleReplacementPlugin } from 'webpack'
import extensions from 'extensions'

export default (neatoConfig: NeatoConfigType): Object => ({
  babel: {
    babelrc: path.join(neatoConfig.projectPath)
  },

  plugins: neatoConfig.action === 'DEV' ? [new HotModuleReplacementPlugin()] : [],

  resolve: {
    extensions: extensions.list.javascript
  },

  module: {
    rules: [
      {
        test: extensions.test.javascript,
        include: [
          path.join(neatoConfig.projectPath, 'src')
        ],
        use: 'babel-loader'
      }
    ]
  }
})
