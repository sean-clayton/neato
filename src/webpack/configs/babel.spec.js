/* eslint-env jest */

import babel from './babel'

describe('babel config', () => {
  xit('Generates a babel-loader config', () => {
    const neatoMock = {
      neatoPath: '/test/neato/path',
      projectPath: '/test/project/path',
      optimize: false,
      options: {
        pages: ['index', 'other-page']
      },
      action: 'DEV'
    }
    expect(babel(neatoMock)).toMatchSnapshot()
  })
})
