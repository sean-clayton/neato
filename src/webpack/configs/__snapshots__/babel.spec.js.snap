exports[`babel config Generates a babel-loader config 1`] = `
Object {
  "babel": Object {
    "babelrc": "/test/project/path",
  },
  "module": Object {
    "rules": Array [
      Object {
        "include": Array [
          "/test/project/path/src",
        ],
        "test": /\.(jsx?|es6)$/,
        "use": "babel-loader",
      },
    ],
  },
  "plugins": Array [
    HotModuleReplacementPlugin {
      "fullBuildTimeout": 200,
      "multiStep": undefined,
    },
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".jsx",
      ".es6",
    ],
  },
}
`;
